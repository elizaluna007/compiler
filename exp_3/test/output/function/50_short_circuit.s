	.text
	.comm	g,4,4
	.global	global
	.type	global, @function
global:
	addi	sp,sp,-8
	sw	ra,0(sp)
	lw	t3,4(sp)
	li	t3,0
	sw	t3,4(sp)
	lw	ra,0(sp)
	addi	sp,sp,8
	jr	ra
	.global	func
	.type	func, @function
func:
	addi	sp,sp,-4
	sw	ra,0(sp)
	call	putint
	la	a0,g
	lw	a0,0(a0)
	lw	ra,0(sp)
	addi	sp,sp,4
	jr	ra
	.global	main
	.type	main, @function
main:
	addi	sp,sp,-8
	sw	ra,0(sp)
	call	global
	lw	t3,4(sp)
	li	t3,0
	sw	t3,4(sp)
	call	getint
	call	func
	call	getint
	call	func
	call	getint
	call	func
	call	getint
	call	func
	call	func
	call	func
	li	a0,0
	lw	ra,0(sp)
	addi	sp,sp,8
	jr	ra
