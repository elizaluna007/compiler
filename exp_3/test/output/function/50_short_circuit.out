11101009900
0